### PR TITLE
refactor: use account's free balance if maximum_collateral is None

### PR DESCRIPTION
**Pull request by gregdhill**
Signed-off-by: Gregory Hill <gregorydhill@outlook.com>

### CODE REVIEW

**Review by sander2 — APPROVED** [review]

File: vault/src/collateral.rs [hunk 1]
```diff
@@ -48,19 +48,25 @@ pub async fn maintain_collateralization_rate(
...
-    )
-    .await?;
+    let actual_collateral = vault.backing_collateral;
+
+    let (required_collateral, maximum_collateral) =
+        future::try_join(provider.get_required_collateral_for_vault(vault_id), async {
+            if let Some(max) = maximum_collateral {
+                Ok(max)
+            } else {
+                provider.get_free_balance().await
```
Comment (sander2) [inline vault/src/collateral.rs:66]: This would leave the vault without funds for paying transaction costs

**Review by sander2 — CHANGES_REQUESTED** [review]
We should probably leave some dot for fees

File: vault/src/collateral.rs (hunk 1 above)
Comment (gregdhill) [reply vault/src/collateral.rs:66]: Note that we wouldn't necessarily use the whole available free balance if it isn't required to maintain secure collateralization.

Comment (gregdhill) [reply vault/src/collateral.rs:66]: I don't see a particularly good way to account for this, we have the same problem is `max_collateral` is too high.

Comment (sander2) [reply vault/src/collateral.rs:66]: It's problematic.. I'm ok with merging now. Can you make a ticket for the issue though? It's definitely something we should look into

**Review by sander2 — APPROVED** [review]

**Pull request merged**